conf/openshift/openshift.json

```diff
@@ -0,0 +1,237 @@
+{
+  "kind": "Template",
+  "apiVersion": "v1",
+  "metadata": {
+    "name": "dataverse",
+    "labels": {
+      "name": "dataverse"
+    },
+    "annotations": {
+      "openshift.io/description": "Dataverse is open source research data repository software: https://dataverse.org",
+      "openshift.io/display-name": "Dataverse"
+    }
+  },
+  "objects": [
+    {
+      "kind": "Service",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "dataverse-glassfish-service"
+      },
+      "spec": {
+        "selector": {
+          "name": "iqss-dataverse-glassfish"
+        },
+        "ports": [
+          {
+            "name": "web",
+            "protocol": "TCP",
+            "port": 8080,
+            "targetPort": 8080
+          }
+        ]
+      }
+    },
+    {
+      "kind": "ImageStream",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "dataverse-plus-glassfish"
+      },
+      "spec": {
+        "dockerImageRepository": "iqss/dataverse-glassfish"
+      }
+    },
+    {
+      "kind": "ImageStream",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "centos-postgresql-94-centos7"
+      },
+      "spec": {
+        "dockerImageRepository": "centos/postgresql-94-centos7"
+      }
+    },
+    {
+      "kind": "ImageStream",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "iqss-dataverse-solr"
+      },
+      "spec": {
+        "dockerImageRepository": "iqss/dataverse-solr"
+      }
+    },
+    {
+      "kind": "DeploymentConfig",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "deploy-dataverse-glassfish",
+        "annotations": {
+          "template.alpha.openshift.io/wait-for-ready": "true"
+        }
+      },
+      "spec": {
+        "template": {
+          "metadata": {
+            "labels": {
+              "name": "iqss-dataverse-glassfish"
+            }
+          },
+          "spec": {
+            "containers": [
```

Comment:
I still really want to see these split out into separate pods.

Reply:
@danmcp can you please link me to an example that's closer to what you want? Thanks!

Reply:
@pdurbin In this example:

https://github.com/openshift/origin/blob/master/examples/sample-app/application-template-stibuild.json


There is the mysql container inside its deployment config:

https://github.com/openshift/origin/blob/master/examples/sample-app/application-template-stibuild.json#L400



And here is the ruby container inside its deployment config:

https://github.com/openshift/origin/blob/master/examples/sample-app/application-template-stibuild.json#L244

Reply:
@danmcp thanks. I made myself a couple screenshots so I can better see what you're talking about. It sounds like you want more than one `DeploymentConfig` entries, each with a single container like this:

![multiplepods](https://user-images.githubusercontent.com/21006/31406601-015b727e-add0-11e7-8d26-94fbe45e944e.png)

What I'm doing right now is only using a single `DeploymentConfig` and putting all three of my containers in it like this:

![singlepod](https://user-images.githubusercontent.com/21006/31406625-160e47dc-add0-11e7-8e58-3bb7ad4e5bba.png)

@danmcp is there a way to have one `DeploymentConfig` run before the other? Right now the Glassfish/Dataverse container restarts until the PostgreSQL and Solr containers are available.